### PR TITLE
updated constraints and removed pre 1.0 legacy section

### PR DESCRIPTION
**Pull request by matthewhelmke**
Do not merge yet--I'm waiting for an answer as to whether to keep the existing `networks` constraint info or whether that constraint is no longer valid. Once answered, I will either update or we can merge as-is.

### CODE REVIEW

**Comment by matthewhelmke** [comment]
Will eventually fix #1249

File: src/en/reference-constraints.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
-Title: Juju constraints  
+Title: Juju constraints
 
 # Constraints
 
-Constraints constrain the possible instances that may be started by juju
+Constraints set limits on the possible instances that may be started by juju
```
Comment (evilnick) [inline src/en/reference-constraints.md]: Juju

File: src/en/reference-constraints.md [hunk 2]
```diff
@@ -19,14 +19,14 @@ environment, no machine will be provisioned, and an error will be printed in the
 machine's entry in juju status.
 
 Constraint defaults can be set on an environment or on specific services by
-using the set-constraints command (see juju help set-constraints).  Constraints
+using the set-constraints command (see juju help set-constraints). Constraints
```
Comment (evilnick) [inline src/en/reference-constraints.md]: backtick the command

File: src/en/reference-constraints.md [hunk 3]
```diff
@@ -19,14 +19,14 @@ environment, no machine will be provisioned, and an error will be printed in the
...
 set on the environment or on a service can be viewed by using the get-
-constraints command.  In addition, you can specify constraints when executing a
+constraints command. In addition, you can specify constraints when executing a
 command by using the --constraints flag (for commands that support it).
```
Comment (evilnick) [inline src/en/reference-constraints.md]: I would probably backtick '--constraints' too

File: src/en/reference-constraints.md [hunk 4]
```diff
@@ -19,14 +19,14 @@ environment, no machine will be provisioned, and an error will be printed in the
...
 
 Constraints specified on the environment and service will be combined to
 determine the full list of constraints on the machine(s) to be provisioned by
-the command.  Service-specific constraints will override environment-specific
+the command. Application-specific constraints will override environment-specific
```
Comment (evilnick) [inline src/en/reference-constraints.md]: :cake:

File: src/en/reference-constraints.md [hunk 5]
```diff
@@ -109,33 +109,56 @@ multiple constraints delimited by a space.
...
 
-In pre-1.0 juju some additional or differently named constraints were
-also supported, these need to be migrated when upgrading.
+    Specifies the type of virtualization to be used, such as `kvm`.
 
-- cpu
 
-    Number of CPU cores for most providers, but equivalent to an Amazon
-    ECU on AWS. Use `cpu-cores` instead.
+## Provider differences
```
Comment (evilnick) [inline src/en/reference-constraints.md]: can we talk about clouds rather than providers.

File: src/en/reference-constraints.md [hunk 6]
```diff
@@ -109,33 +109,56 @@ multiple constraints delimited by a space.
...
-- ec2-zone
+Different providers support different constraints and sometimes different
+values for these constraints. Sometimes, different providers also dictate
+constraints that would conflict with other providers and cannot be used
+in combination. Use this list to help you understand the differing needs.
 
-    EC2 availability zone that a service unit must be deployed into. No
-    equivalent implemented as of juju 1.12, follow [bug
-    1183831](https://bugs.launchpad.net/juju-core/+bug/1183831).
+###Azure Provider:
```
Comment (evilnick) [inline src/en/reference-constraints.md]: This is better than not having the information, but it still feels rather clunky to use and understand. Maybe for now it is sufficient to add a TODO to the doc and file a new issue (medium priority)

**Comment by evilnick** [comment]
I think that's the end of my input here - good work on spotting the MAAS and service bits :)

**Comment by matthewhelmke** [comment]
Thanks! I'll update what we have so far in the PR.

**Comment by matthewhelmke** [comment]
Fixes #1249

**Comment by nskaggs** [comment]
This doesn't appear to be live yet -- when will it be?

**Comment by evilnick** [comment]
It is live in the /dev docs - https://jujucharms.com/docs/devel/reference-constraints

I was waiting to see if there was an idea when the cpu-core -> core change will land, but it doesn't look like it will be soon. In which case we will put a note on it and push it out to 2.0

**Pull request merged**